ui/src/components/shared/SiderMenu.tsx

```diff
@@ -35,12 +40,19 @@ export function SiderMenu() {
           items={[
             {
               icon: <IconSchema />,
+              // TODO - these keys need to be typed.
```

Comment:
I assume the problem you want to solve is a key collision, right? If that is the case, I'm not getting how a union type would help there, since the type would not save us from:

 * Using any other arbitrary string as key
 * Using repeated values that honor the type
 
Or are you maybe solving another problem that I have not thought of?

Reply:
Thanks for helping come up with the optimal solution!